core/transaction/transaction.go

```diff
@@ -101,7 +101,9 @@ func (tx *Transaction) SerializeUnsigned(w io.Writer) error {
 	}
 	tx.Payload.Serialize(w)
 	//nonce
-	serialization.WriteVarUint(w, tx.Nonce)
+	if len(tx.BalanceInputs) > 0 {
```

Comment:
why the Nonce not be serialization when balance input > 0 and what the relative handle in deserialization  function?